src/rules/isort/split.rs

```diff
@@ -0,0 +1,40 @@
+use std::collections::BTreeSet;
+
+use crate::rules::isort::types::{ImportBlock, Importable};
+
+pub fn split_by_forced_separate<'a>(
+    block: ImportBlock<'a>,
+    forced_separate: &BTreeSet<String>,
+) -> [ImportBlock<'a>; 2] {
+    let (forced_separate_import, regular_import) = block
+        .import
+        .into_iter()
+        .partition(|(import, _)| forced_separate.contains(&import.module_base()));
+    let (forced_separate_import_from, regular_import_from) = block
+        .import_from
+        .into_iter()
+        .partition(|(import_from, ..)| forced_separate.contains(&import_from.module_base()));
+    let (forced_separate_import_from_as, regular_import_from_as) = block
+        .import_from_as
+        .into_iter()
+        .partition(|((import_from, _), ..)| forced_separate.contains(&import_from.module_base()));
+    let (forced_separate_import_from_star, regular_import_from_star) = block
+        .import_from_star
+        .into_iter()
+        .partition(|(import_from, ..)| forced_separate.contains(&import_from.module_base()));
+
+    [
+        ImportBlock {
+            import: regular_import,
+            import_from: regular_import_from,
+            import_from_as: regular_import_from_as,
+            import_from_star: regular_import_from_star,
+        },
+        ImportBlock {
+            import: forced_separate_import,
+            import_from: forced_separate_import_from,
+            import_from_as: forced_separate_import_from_as,
+            import_from_star: forced_separate_import_from_star,
+        },
```

Comment:
If you had this:

```py
import os
import foo
import bar
import typing
```

And `foo` and `typing` were in `forced_separate`, what's the right behavior?

Reply:
As in: what does `isort` do, and are we consistent?

Reply:
Ah, duh... Actually, based on [the test in `isort`](https://github.com/PyCQA/isort/blob/82569d97210a487786cab5dae9e5ce0c0c9af82f/tests/unit/test_isort.py#L1039-L1099) we should partition the blocks _separately_ for each entry in `forced_separate`, and take the order of `forced_separate` into account too.

Back to the drawing board... :)

Reply:
Okay, rewrote the splitting to partition into any number of separate blocks based on the order of `forced_separate`.